map/src/test/java/openlr/map/simplemockdb/SimpleMockedLine.java

```diff
@@ -0,0 +1,231 @@
+     */
+    @Deprecated
+    public Point2D.Double getPointAlongLine(int var1) {
+        return null;
+    }
+
+    public GeoCoordinates getGeoCoordinateAlongLine(int distance) throws SimpleMockedException {
+
+        int lengthCovered = 0;
+        for (LineSegment segment : lineSegments) {
+            if (lengthCovered <= distance && lengthCovered + segment.getLength() >= distance) {
+                double offset = distance - lengthCovered;
+                Coordinate point = segment.pointAlong(segment.getLength() / offset);
+                point = segment.project(point);
+                double z = segment.p0.z + 1 * (segment.p1.z - segment.p0.z);
+                point.setOrdinate(2, z);
+                return ToWGS84(point);
+            }
+            lengthCovered += segment.getLength();
+        }
+        throw new SimpleMockedException("length is greater than the line length");
+    }
+
+    public int getLineLength() {
+        return (int) this.lineSegments.stream().mapToDouble(segment -> segment.getLength()).sum();
+    }
+
+    public long getID() {
+        return xmlLine.getId().longValue();
+    }
+
+    public Iterator<Line> getPrevLines() {
+        return this.startNode.getIncomingLines();
+    }
+
+    /**
+     * @deprecated
+     */
+    @Deprecated
+    public java.awt.geom.Path2D.Double getShape() {
+        return null;
+    }
+
+    public Iterator<Line> getNextLines() {
+        return this.endNode.getOutgoingLines();
+    }
+
+    public boolean equals(Object var1) {
+        if (var1 instanceof SimpleMockedLine) {
+            return this.getID() == ((SimpleMockedLine) var1).getID();
+        }
+        return false;
+    }
+
+    public int hashCode() {
+        return this.xmlLine.hashCode();
+    }
+
+    public int distanceToPoint(double var1, double var3) {
+        Geometry geometry = factory.createPoint(toCartesian(var1, var3));
+        return (int) Math.round(getLineString().distance(geometry));
+    }
+
+    public int measureAlongLine(double lon, double lat) {
+        Coordinate crd = toCartesian(lon, lat);
+        int segmentLength = 0;
+        for (LineSegment segment : lineSegments) {
+            if (segment.distance(crd) == 0) {
+                Coordinate start = segment.p0;
+                return (int) (segmentLength + (new LineSegment(start, crd)).getLength());
+            }
+            segmentLength += segment.getLength();
+        }
+        throw new SimpleMockedException("point is not on the line");
+    }
+
+    public List<GeoCoordinates> getShapeCoordinates() {
+        List<GeoCoordinates> shape = new ArrayList<>();
+        for (int index = 0; index < lineSegments.size(); ++index) {
+            shape.add(ToWGS84(lineSegments.get(index).p0));
+
+            if (index == lineSegments.size() - 1) {
+                shape.add(ToWGS84(lineSegments.get(index).p1));
+            }
+        }
+        return shape;
+    }
+
+    public Map<Locale, List<String>> getNames() {
```

Comment:
You should not return `null` when you can return an empty Map.

Reply:
9921bc46eaef1f2b022379c9e24cba52db5b1735